lib/rig_inbound_gateway_web/v1/metadata_controller.ex

```diff
@@ -0,0 +1,178 @@
+
+  ## Dirty testing without JWT
+      META='{ "metadata": { "userid": "9ab1bff2-a8d8-455c-b48a-50145d7d8e30", "locale": "de-AT", "timezone": "GMT+2" } }'
+      http put ":4000/_rig/v1/connection/sse/${CONN_TOKEN}/metadata" <<< "$META"
+
+  """
+  @spec set_metadata(conn :: Plug.Conn.t(), params :: map) :: Plug.Conn.t()
+  def set_metadata(
+        %{method: "PUT"} = conn,
+        %{
+          "connection_id" => connection_id
+        }
+      ) do
+
+    conn
+    |> accept_only_req_for(["application/json"])
+    |> decode_metadata()
+    |> do_set_metadata(connection_id)
+  end
+
+  defp decode_metadata(conn) do
+    with {"application", "json"} <- content_type(conn),
+         {:ok, body, conn} <- BodyReader.read_full_body(conn),
+         {:ok, json} <- Jason.decode(body),
+         {:parse, %{"metadata" => metadata}} <- {:parse, json} do
+
+      conf = config()
+      jwt_fields = conf.jwt_fields
+
+      with {:ok, auth_tokens} <- Map.get(conn.assigns, :auth_tokens, [])
+      |> Enum.map(fn
+        {"bearer", token} -> JWT.parse_token(token)
+        _ -> {:ok, %{}}
+      end)
+      |> Result.list_to_result()
+      |> Result.map_err(fn decode_errors -> {:error, decode_errors} end) do
+
+        # Merge the list of maps into a single map
+        auth_tokens = Enum.reduce(auth_tokens, fn x, y ->
+          Map.merge(x, y, fn _k, v1, v2 -> v2 ++ v1 end)
+        end)
+
+        # Get values from JWT
+        jwt_vals = jwt_fields
+        |> Enum.map(fn x ->
+          key = x |> elem(0)
+          jwt_key = x |> elem(1)
+          {key, auth_tokens[jwt_key]}
+        end)
+        |> Enum.into(%{})
+
+        # Merge values from metadata with the values from JWT, prioritizing JWT values
+        metadata = Map.merge(metadata, jwt_vals)
+
+        conn
+        |> Plug.Conn.assign(:metadata, metadata)
+      else
+        {:error, _} ->
+          conn
+          |> Plug.Conn.assign(:metadata, metadata)
+      end
+    else
+      {:parse, json} ->
+        message = """
+        Expected field "metadata" is not present.
+
+        Decoded request body:
+        #{inspect(json)}
+        """
+
+        fail!(conn, message)
+
+      error ->
+        message = """
+        Expected JSON encoded body.
+
+        Technical info:
+        #{inspect(error)}
+        """
+
+        fail!(conn, message)
+    end
+  end
+
+  defp do_set_metadata(%{halted: true} = conn, _connection_id), do: conn
+
+  defp do_set_metadata(%{assigns: %{metadata: metadata}} = conn, connection_id) do
+    # Check if metadata contains all the indexed fields; if not: send back a bad request
+    # This needs to be done here because now the values from the JWT got inserted
+    if contains_idx?(metadata) === false do
+      message = """
+        Metadata doesn't contain indexed fields.
+        """
+
+      fail!(conn, message)
+    else
+      Logger.info("Metadata: " <> inspect(metadata))
```

Comment:
* Please use `info` level only for important events, not for debugging metadata like this.
* When using `Logger`, _always_ use a function argument. E.g.: `Logger.info(fn -> "foo bar" end)`